src/c2s/mongoose_c2s.erl

```diff
@@ -705,13 +706,16 @@ handle_state_result(StateData0, C2SState, MaybeAcc,
     maybe_send_xml(StateData2, MaybeAcc, MaybeSocketSend),
     {next_state, NextFsmState, StateData2, MaybeActions}.
 
+-spec maybe_send_xml(c2s_data(), mongoose_acc:t(), [exml:element()]) -> ok.
 maybe_send_xml(_StateData, _Acc, []) ->
     ok;
 maybe_send_xml(StateData = #c2s_data{host_type = HostType, lserver = LServer}, undefined, ToSend) ->
     Acc = mongoose_acc:new(#{host_type => HostType, lserver => LServer, location => ?LOCATION}),
-    send_element(StateData, ToSend, Acc);
+    [send_element(StateData, El, Acc) || El <- ToSend],
```

Comment:
And here we're losing an optimisation. We could call once per packet `exml:to_iolist/1`, which will generate a single binary, and then `gen_tcp:send/X`, which will do one socket operation, as many times as there are packets available; or, we could do a single call to `exml:to_iolist/1` with the list of records, which will return a iolist list of binaries, and do a single socket operation with that iolist.

Considering that encoding and socket operations are more expensive than metrics, I'd keep them in their most optimal usage and have metrics either do `iolist_size/1`, which will perfectly free anyway, or, have the metrics code explore the list of binaries and do the update per metrics.